lib/cache/cache.go

```diff
@@ -235,6 +235,11 @@ func ForNode(cfg Config) Config {
 		{Kind: types.KindNamespace, Name: apidefaults.Namespace},
 		{Kind: types.KindNetworkRestrictions},
 	}
+
+	if cfg.DiscoveryAgent {
+		cfg.Watches = append(cfg.Watches, types.WatchKind{Kind: types.KindNode})
```

Comment:
How about we make a separate `DiscoveryAccessPoint` interface that will have `GetNodes` method and will only get initialized in the server via the server option ([here](https://github.com/gravitational/teleport/blob/master/lib/service/service.go#L2257)) if discovery is enabled. Then it can return a proper error message if a node that does not have discovery enabled tries to fetch the nodes. Then we should also probably do separate `ForDiscovery` cache setup above rather than adding it to `ForNode`.